lib/gems/pending/openstack/openstack_handle/pagination/marker.rb

```diff
@@ -16,7 +16,7 @@ def list
           all_objects.concat(objects_on_page)
         end
 
-        all_objects
+        all_objects.uniq(&:identity)
```

Comment:
This change is not the fix for any issue, duplication here seems unrelevant. Tested return
```all_objects + all_objects``` with no change (no duplicated records in the app).

Reply:
@aufi so, can you test this scenario: "fog it returns duplicates once, then it returns uniq"

That is what I think is happening. If parser returns duplicates, we should see it twice in the DB. When parser returns uniq, we disconnect 1 of the duplicates. In this case, only 1 of the duplicates is associated to the Service.

-------

the problematic part is https://github.com/Ladas/manageiq/blob/3ec67f4d8059a1846c7a707b7ad4f589ca6a8f14/app/models/ems_refresh/save_inventory.rb#L66
then https://github.com/Ladas/manageiq/blob/3ec67f4d8059a1846c7a707b7ad4f589ca6a8f14/app/models/ems_refresh/save_inventory.rb#L126

Reply:
@Ladas I tested that scenario and it works the way you expected. Trimming out the duplicates from fog or discarding them during refresh/save works, but leaves any disconnected duplicates that have already been created.

Reply:
@mansam right, so we should provide a rails c scrips for cleanup of the duplicates (since we can't backport a migration). And we are preparing migrations for the unique indexes, so it there, we should think about cleaning up these duplicates